yearn/v2/strategies.py

```diff
@@ -96,8 +100,96 @@ def load_harvests(self):
 
     @property
     def harvests(self) -> List[int]:
-        self.load_harvests()
+        if len(self._harvests) == 0:
+            self.load_harvests()
         return self._harvests
+        
+    @property
+    def harvests_data(self):
+        if len(self._harvests) == 0:
+            self.load_harvests()
+        sorted_harvests = sorted(self._harvests_data, key=lambda k: k['block_number'], reverse=True)
+        return sorted_harvests
+
+    @property
+    def debt_ratio(self):
+        return self.vault.vault.strategies(self.strategy.address).dict()['debtRatio'] / 1e4
+
+    @property
+    def apy(self) -> StrategyApy:
+        harvests = self.harvests_data
+
+        # Find at least two profitable harvests
+        profitable_harvests_count = 0
+        for idx, harvest in enumerate(harvests):
+            profit = harvest['event']['profit']
+            if profit > 0:
+                profitable_harvests_count += 1
+                if profitable_harvests_count == 1:
+                    latest_harvest_with_profit_idx = idx
+                elif profitable_harvests_count == 2:
+                    second_latest_harvest_with_profit_idx = idx
+                    break
+
+        # Not enough profitable harvests
+        if profitable_harvests_count < 2:
+            return StrategyApy(0, 0)
+
+        # Not enough data
+        have_enough_data = second_latest_harvest_with_profit_idx + 1 < len(harvests)
+        if have_enough_data == False:
+            return StrategyApy(0, 0)
+
+        # Latest profitable harvest
+        latest_harvest_with_profit_current = harvests[latest_harvest_with_profit_idx]
```

Comment:
this part is pretty hard to read and debug imo. Consider adding some helper methods and return a tuple or a dataclass.
E.g.
```
first_harvest_current, second_harvest_current, duration_current = _get_profitable_harvests_with_duration(harvests, idx+1)
first_harvest_previous, second_harvest_previous, duration_previous = _get_profitable_harvests_with_duration(harvests, idx)
```

Another good option would be to use a dataclass for the return value and then use it for the apr calculations so it becomes:
```
current = _get_harvests_with_duration(idx+1)
previous = _get_harvests_with_duration(idx)
```

then further below it becomes:
```
current_apr = current.profit / ...
```